staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/validation/validation.go

```diff
@@ -162,144 +162,174 @@ func ValidateCustomResourceDefinitionNames(names *apiextensions.CustomResourceDe
 	return allErrs
 }
 
+// additionalSpecValidator applies validations for different OpenAPI specfication versions.
+type additionalSpecValidator interface {
+	validateCustomResourceDefinitionOpenAPISpec(spec *apiextensions.JSONSchemaProps, fldPath *field.Path) field.ErrorList
+}
+
 // ValidateCustomResourceDefinitionValidation statically validates
-func ValidateCustomResourceDefinitionValidation(CustomResourceValidation *apiextensions.CustomResourceValidation, fldPath *field.Path) field.ErrorList {
+func ValidateCustomResourceDefinitionValidation(customResourceValidation *apiextensions.CustomResourceValidation, fldPath *field.Path) field.ErrorList {
 	allErrs := field.ErrorList{}
 
-	if CustomResourceValidation.OpenAPISpecV2 != nil && CustomResourceValidation.OpenAPISpecV3 != nil {
+	if customResourceValidation.OpenAPISpecV2 != nil && customResourceValidation.OpenAPISpecV3 != nil {
 		allErrs = append(allErrs, field.Forbidden(fldPath.Child(""), "only one openAPISpec can be specified."))
 	}
 
-	if CustomResourceValidation.OpenAPISpecV2 != nil {
-		allErrs = append(allErrs, ValidateCustomResourceDefinitionOpenAPISpecV2(CustomResourceValidation.OpenAPISpecV2, fldPath.Child("openAPISpecV2"))...)
+	if customResourceValidation.OpenAPISpecV2 != nil {
+		openAPISpecV2 := &additionalSpecValidatorV2{}
+		allErrs = append(allErrs, ValidateCustomResourceDefinitionOpenAPISpec(customResourceValidation.OpenAPISpecV2, fldPath.Child("openAPISpecV2"), openAPISpecV2)...)
 	}
 
-	if CustomResourceValidation.OpenAPISpecV3 != nil {
-		allErrs = append(allErrs, ValidateCustomResourceDefinitionOpenAPISpec(CustomResourceValidation.OpenAPISpecV3, fldPath.Child("openAPISpecV3"))...)
+	if customResourceValidation.OpenAPISpecV3 != nil {
+		openAPISpecV3 := &additionalSpecValidatorV3{}
+		allErrs = append(allErrs, ValidateCustomResourceDefinitionOpenAPISpec(customResourceValidation.OpenAPISpecV3, fldPath.Child("openAPISpecV3"), openAPISpecV3)...)
 	}
 
 	return allErrs
 }
 
-// ValidateCustomResourceDefinitionOpenAPISpecV2 statically validates
-func ValidateCustomResourceDefinitionOpenAPISpecV2(OpenAPISpecV2 *apiextensions.JSONSchemaProps, fldPath *field.Path) field.ErrorList {
+// ValidateCustomResourceDefinitionOpenAPISpec statically validates
+func ValidateCustomResourceDefinitionOpenAPISpec(schema *apiextensions.JSONSchemaProps, fldPath *field.Path, specVersion additionalSpecValidator) field.ErrorList {
```

Comment:
`specVersion` is a misleading name. Would it better to be called `versionSpecificValidator` or simply `additionalValidator`?